#pruebasUnitarias 

## Interpretación de la salida de `pytest`:

#### 1. **Punto (`.`) para pruebas exitosas:**
    
    - Cada punto (`.`) que aparece en la salida significa que una prueba se ejecutó correctamente (pasó sin errores).
    - Ejemplo:


```c
test_api\test_get_all_users.py .
test_api\test_usuarios.py ...
```

Aquí, cada `.` representa una prueba exitosa. En total, pasaron 4 pruebas: una en `test_get_all_users.py` y tres en `test_usuarios.py`.

#### 2. **Letra `F` para pruebas fallidas:**


- Si aparece una `F` en la salida, significa que una prueba falló.
- Ejemplo:

```c
test_api\test_get_all_users.py .F 
```

#### 3. **Detalles sobre los fallos:**

- Cuando una prueba falla, `pytest` te dará más detalles justo después de la línea con la `F`. Puedes ver exactamente qué salió mal y por qué.
- Ejemplo de fallo:

```c
============================================================================ FAILURES ============================================================================
_________________________________________________________________________ test_get_all_users_not_found _________________________________________________________________________ 

def test_get_all_users_not_found():
    response = client.get("/usuarios")
>       assert response.status_code == 404
E       assert 200 == 404
E        +  where 200 = <Response [200 OK]>.status_code

test_api\test_get_all_users.py:16: AssertionError

```

En este caso, la prueba **`test_get_all_users_not_found`** falló porque esperaba un código de estado 404 (usuario no encontrado), pero recibió un 200 (éxito).

### Resumen de pruebas al final:

Al final de la ejecución de las pruebas, `pytest` muestra un resumen como este:

```c
collected 4 items

test_api\test_get_all_users.py .F                                                                                                                                      [ 50%]
test_api\test_usuarios.py ...                                                                                                                                          [100%]

=================================================================================== FAILURES ==================================================================================== 
_________________________________________________________________________ test_get_all_users_not_found __________________________________________________________________________ 
...

========================================================================= 1 failed, 3 passed in 1.07s ========================================================================== 

```

### Prueba fallida:

![[Pasted image 20241009080304.png]]

### Prueba exitosa:

![[Pasted image 20241009080336.png]]
